src/lib/field-next/field-foundation.ts

```diff
@@ -0,0 +1,268 @@
+import { ICustomElementFoundation } from '@tylertech/forge-core';
+import { IFieldAdapter } from './field-adapter';
+import { FieldDensity, FieldLabelAlignment, FieldLabelPosition, FieldShape, FieldSlot, FieldSupportTextInset, FieldTheme, FieldVariant, FIELD_CONSTANTS } from './field-constants';
+
+export interface IFieldFoundation extends ICustomElementFoundation {
+  labelPosition: FieldLabelPosition;
+  labelAlignment: FieldLabelAlignment;
+  floatLabel: boolean;
+  invalid: boolean;
+  required: boolean;
+  optional: boolean;
+  disabled: boolean;
+  variant: FieldVariant;
+  theme: FieldTheme;
+  shape: FieldShape;
+  density: FieldDensity;
+  dense: boolean;
+  popoverIcon: boolean;
+  popoverExpanded: boolean;
+  multiline: boolean;
+  supportTextInset: FieldSupportTextInset;
+  initialize(): void;
+}
+
+export class FieldFoundation implements IFieldFoundation {
+  private _labelPosition: FieldLabelPosition = FIELD_CONSTANTS.defaults.DEFAULT_LABEL_POSITION;
+  private _labelAlignment: FieldLabelAlignment = FIELD_CONSTANTS.defaults.DEFAULT_LABEL_ALIGNMENT;
+  private _floatLabel = false;
+  private _invalid = false;
+  private _required = false;
+  private _optional = false;
+  private _disabled = false;
+  private _variant: FieldVariant = FIELD_CONSTANTS.defaults.DEFAULT_VARIANT;
+  private _theme: FieldTheme = FIELD_CONSTANTS.defaults.DEFAULT_THEME;
+  private _shape: FieldShape = FIELD_CONSTANTS.defaults.DEFAULT_SHAPE;
+  private _density: FieldDensity = FIELD_CONSTANTS.defaults.DEFAULT_DENSITY;
+  private _dense = false;
+  private _popoverIcon = false;
+  private _popoverExpanded = false;
+  private _multiline = false;
+  private _supportTextInset: FieldSupportTextInset = FIELD_CONSTANTS.defaults.DEFAULT_SUPPORT_TEXT_INSET;
+  private _isInitialized = false;
```

Comment:
You should try replacing this property with `this._adapter.isConnected` instead unless it relies on timing of internal setup instead of just being attached to the DOM.